contracts/EtomicSwapNft.sol

```diff
@@ -0,0 +1,719 @@
+            );
+        }
+
+        takerPayments[id].state = TakerPaymentState.TakerRefunded;
+
+        emit TakerPaymentRefundedTimelock(id);
+
+        uint256 total_amount = amount + dexFee;
+        if (tokenAddress == address(0)) {
+            payable(msg.sender).transfer(total_amount);
+        } else {
+            IERC20 token = IERC20(tokenAddress);
+            token.safeTransfer(msg.sender, total_amount);
+        }
+    }
+
+    function refundTakerPaymentSecret(
+        bytes32 id,
+        uint256 amount,
+        uint256 dexFee,
+        address maker,
+        bytes32 takerSecret,
+        bytes32 makerSecretHash,
+        address tokenAddress
+    ) external {
+        require(
+            takerPayments[id].state == TakerPaymentState.PaymentSent,
+            "Invalid payment state. Must be PaymentSent"
+        );
+
+        bytes20 paymentHash = ripemd160(
+            abi.encodePacked(
+                amount,
+                dexFee,
+                maker,
+                msg.sender,
+                sha256(abi.encodePacked(takerSecret)),
+                makerSecretHash,
+                tokenAddress
+            )
+        );
+
+        require(
+            paymentHash == takerPayments[id].paymentHash,
+            "Invalid paymentHash"
+        );
+
+        takerPayments[id].state = TakerPaymentState.TakerRefunded;
+
+        emit TakerPaymentRefundedSecret(id, takerSecret);
+
+        uint256 total_amount = amount + dexFee;
+        if (tokenAddress == address(0)) {
+            payable(msg.sender).transfer(total_amount);
+        } else {
+            IERC20 token = IERC20(tokenAddress);
+            token.safeTransfer(msg.sender, total_amount);
+        }
+    }
+
+    struct HTLCParams {
+        bytes32 id;
+        address taker;
+        address tokenAddress;
+        bytes32 takerSecretHash;
+        bytes32 makerSecretHash;
+        uint32 paymentLockTime;
+    }
+
+    function onERC1155Received(
+        address operator,
+        address from,
+        uint256 tokenId,
+        uint256 value,
+        bytes calldata data
+    ) external override returns (bytes4) {
+        // Decode the data to extract HTLC parameters
+        HTLCParams memory params = abi.decode(data, (HTLCParams));
+
+        require(params.taker != address(0), "Taker must not be zero address");
+        require(
+            params.tokenAddress != address(0),
+            "Token must not be zero address"
+        );
+        require(
+            msg.sender == params.tokenAddress,
+            "Token address does not match sender"
+        );
+        require(operator == from, "Operator must be the sender");
+        require(value > 0, "Value must be greater than 0");
+        require(
+            makerPayments[params.id].state == MakerPaymentState.Uninitialized,
+            "Maker ERC1155 payment must be Uninitialized"
+        );
+        require(!isContract(params.taker), "Taker cannot be a contract");
```

Comment:
Order of requires is changed from ETH/ERC20 contract, shouldn't
```js
require(
            makerPayments[params.id].state == MakerPaymentState.Uninitialized,
            "Maker ERC1155 payment must be Uninitialized"
        );
```
be first?

Reply:
You are right! done